### PR TITLE
Plotting improvements

### PR DESCRIPTION
**Pull request by RJTK**
I updated the plotting interface to allow passing in a figure.  e.g.

```
np.random.seed(2718)
f_true = gpo.objective_examples.experiments2d.dropwave(
    bounds=[(-5.12, 5.12), (-5.12, 5.12)])
domain = [{'name': 'var_%s' % i, 'type': 'continuous',
           'domain': f_true.bounds[i]} for i in range(2)]

problem = gpo.methods.BayesianOptimization(f=f_true.f,
                                           domain=domain,
                                           acquisition_type='EI',
                                           exact_feval=True)

# Run the optimization
fig = plt.figure()
max_iter = 10  # evaluation budget
max_time = 60  # time budget
eps = 10e-6  # Minimum distance between the last two observations
problem.run_optimization(max_iter, max_time, eps)
problem.plot_acquisition(fig=fig)
f_true.plot(ax=fig.get_axes()[0], plt_type='contour')
fig.get_axes()[0].legend()
fig.show()
plt.show()
```

Will plot on a figure that is owned by the user.  `f_true.plot` is then able to draw contour lines on top of what `problem.plot_acquisition` plotted.

In 1 dimension

```
f_true = gpo.objective_examples.experiments1d.forrester()
bounds = [{'name': 'var_1', 'type': 'continuous',
           'domain': (0, 1)}]

problem = gpo.methods.BayesianOptimization(f=f_true.f,
                                           domain=bounds,
                                           acquisition_type='LCB',
                                           exact_feval=True,
                                           normalize_Y=False)

# Run the optimization
fig = plt.figure()
max_iter = 10  # evaluation budget
max_time = 60  # time budget
eps = 10e-6  # Minimum distance between the last two observations
problem.run_optimization(max_iter, max_time, eps)
problem.plot_acquisition(fig=fig)
f_true.plot(ax=fig.get_axes()[0])
fig.show()
plt.show()
```

The behaviour is similar.  `problem.plot_acquisition` takes in a figure, adds an axis and then plots on that axis.  Possibly I should give an option to accept either a figure or an axis object?  The reason I am passing a figure to `plot_acquisition` is that it needs to create 3 subplots (hence 3 axes) for the 2D plotting.

All the arguments are option, and the default behaviour is roughly the same as before.  But I cleaned up some of the code, removing duplication and erroneous whitespace, and made the plots look a bit (i think) better.

examples (from above):

ei_dropwave

lcb_forrester

### CODE REVIEW

**Comment by apaleyes** [comment]
Thanks for the PR! One immediate comment (this also concerns the other PR as well) - can you please clean up the commit history a bit please? It would be great to have 1 or 2 commits in git history for this scope of changes, as it stands right now it seems too fragmented.

**Comment by RJTK** [comment]
I'm unsure of how to properly clean up the history.  I used `git rebase -i HEAD~11`, then used 'fixup' for to squish together all the previous commits.  I then had to merge something from remote, and used the default "Merge branch 'plotting_improvements' of https://github.com/RJTK/GPyOpt  ..." commit message.  I'm now trying to get rid of that commit message with `git rebase -i HEAD~2`, but am getting the error:

```
First, rewinding head to replay your work on top of it...
Applying: Refactored / improved plot_acquisition
Using index info to reconstruct a base tree...
M	GPyOpt/core/bo.py
M	GPyOpt/plotting/plots_bo.py
Falling back to patching base and 3-way merge...
Auto-merging GPyOpt/plotting/plots_bo.py
CONFLICT (content): Merge conflict in GPyOpt/plotting/plots_bo.py
Failed to merge in the changes.
Patch failed at 0001 Refactored / improved plot_acquisition

When you have resolved this problem, run "git rebase --continue".
If you prefer to skip this patch, run "git rebase --skip" instead.
To check out the original branch and stop rebasing, run "git rebase --abort".
```

**Comment by RJTK** [comment]
I think I fixed the problem I had, but I really butchered the commit history.  Not so sure what I should be doing here...

File: GPyOpt/objective_examples/experiments2d.py [hunk 1]
```diff
@@ -13,36 +13,59 @@
...
-    def plot(self):
+    def plot(self, ax=None, plt_type='contourf'):
         bounds = self.bounds
-        x1 = np.linspace(bounds[0][0], bounds[0][1], 100)
-        x2 = np.linspace(bounds[1][0], bounds[1][1], 100)
-        X1, X2 = np.meshgrid(x1, x2)
-        X = np.hstack((X1.reshape(100*100,1),X2.reshape(100*100,1)))
+        x1 = np.arange(*bounds[0], step=0.01)
+        if len(x1) > 1000:
+            x1 = np.linspace(*bounds[0], num=1000)
```
Comment (apaleyes) [inline GPyOpt/objective_examples/experiments2d.py]: You can use "min" and get rid of "if"

File: GPyOpt/objective_examples/experiments2d.py [hunk 2]
```diff
@@ -13,36 +13,59 @@
...
     '''
-    
-    def plot(self):
+    def plot(self, ax=None, plt_type='contourf'):
         bounds = self.bounds
-        x1 = np.linspace(bounds[0][0], bounds[0][1], 100)
-        x2 = np.linspace(bounds[1][0], bounds[1][1], 100)
-        X1, X2 = np.meshgrid(x1, x2)
-        X = np.hstack((X1.reshape(100*100,1),X2.reshape(100*100,1)))
+        x1 = np.arange(*bounds[0], step=0.01)
```
Comment (apaleyes) [inline GPyOpt/objective_examples/experiments2d.py]: "*bounds[0]" is a neat trick, but let's probably revert to old code here. Old one is much more readable.

File: GPyOpt/plotting/plots_bo.py [hunk 3]
```diff
@@ -2,149 +2,183 @@
...
 
 import numpy as np
-from pylab import grid
 import matplotlib.pyplot as plt
-from pylab import savefig
-import pylab
 
 
-def plot_acquisition(bounds,input_dim,model,Xdata,Ydata,acquisition_function,suggested_sample, filename = None):
+def _normalize_acqu(acqu, m, pct95):
```
Comment (apaleyes) [inline GPyOpt/plotting/plots_bo.py]: Again, let's refraining from unnecessary shortenings. Also this method could use some docstring to explain what it's for.

**Comment by apaleyes** [comment]
@vlavorini no, unfortunately not. We haven't heard from @RJTK for a while, but if someone picks up this PR and fixes the comments above, we will be happy to reconsider merging.

As we haven't heard from the PR's author for a while, I am going to close this.

**Pull request closed without merging**